lib/active_set/sorting/operation.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+require_relative '../attribute_instruction'
+
+class ActiveSet
+  module Sorting
+    class Operation
+      def initialize(set, instructions_hash)
+        @set = set
+        @instructions_hash = instructions_hash
+      end
+
+      def execute
+        attribute_instructions = @instructions_hash
+                                   .flatten_keys
+                                   .map { |k, v| AttributeInstruction.new(k, v) }
+
+        activerecord_strategy = ActiveRecordStrategy.new(@set, attribute_instructions)
+        if activerecord_strategy.executable_instructions == attribute_instructions
+          activerecord_sorted_set = activerecord_strategy.execute
+        end
+
+        return activerecord_sorted_set if attribute_instructions.all?(&:processed?)
+
+        EnumerableStrategy.new(@set, attribute_instructions).execute
+      end
+
+      def operation_instructions
+        @instructions_hash.symbolize_keys
+      end
+    end
+
+    class EnumerableStrategy
+      def initialize(set, attribute_instructions)
+        @set = set
+        @attribute_instructions = attribute_instructions
+      end
+
+      def execute
+        @set.sort_by do |item|
+          @attribute_instructions.map do |instruction|
+            sortable_numeric_for(instruction, item) * direction_multiplier(instruction.value)
+          end
+        end
+      end
+
+      def sortable_numeric_for(instruction, item)
```

Comment:
This is the heart of the EnumerableStrategy for sorting. We convert any item value obtained via an attribute instruction into a numerical value that properly encodes sort order.